File: Cpp/2079_watering_plants.cpp
// https://leetcode.cn/problems/watering-plants/

class Solution {
public:
    int wateringPlants(vector<int>& plants, int capacity) {
        int res = 0, cur = capacity;
        for (int i = 0; i < plants.size(); ++i) {
            if (cur >= plants[i]) {
                cur -= plants[i];
                res += 1;
            }
            else {
                cur = capacity - plants[i];
                res += (i+i+1);
            }
        }
        return res;
    }
};

File: Python3/2079_watering_plants.py
# https://leetcode.cn/problems/watering-plants/

class Solution:
    def wateringPlants(self, plants, capacity):
        """
        Inputs:
            plants: list[int]
            capacity: int
        Outputs:
            res: int
        """
        res = 0
        cur = capacity
        for i, p in enumerate(plants):
            if cur >= p:
                cur -= p
                res += 1
            else:
                cur = capacity - p
                res += (i+i+1)
        return res